### PR TITLE
[formrecognizer] Remove get children methods

### PR DESCRIPTION
**Pull request by catalinaperalta**
This PR removes `get_words` and `get_lines` methods on all models except for `DocumentLine`. 

Update samples to only showcase calling `get_words` on a DocumentLine. 
Update other docs related to this change.

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/samples/README.md [hunk 1]
```diff
@@ -39,7 +39,6 @@ All of these samples need the endpoint to your Form Recognizer resource ([instru
 |[sample_manage_models.py][sample_manage_models] and [sample_manage_models_async.py][sample_manage_models_async]|Manage the models in your account|
 |[sample_get_operations.py][sample_get_operations] and [sample_get_operations_async.py][sample_get_operations_async]|Get and list the document model operations created within the past 24 hours|
 |[sample_copy_model.py][sample_copy] and [sample_copy_model_async.py][sample_copy_async]|Copy a custom model from one Form Recognizer resource to another|
-|[sample_get_document_elements.py][sample_get_document_elements] and [sample_get_document_elements_async.py][sample_get_document_elements_async]|Get document elements, like words and lines, related to other elements returned after an analysis operation|
```
Comment (catalinaperalta) [inline sdk/formrecognizer/azure-ai-formrecognizer/samples/README.md:42]: Note that I will update this in a separate PR

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py [hunk 2]
```diff
@@ -2808,7 +2728,11 @@ def get_words(self, **kwargs):  # pylint: disable=unused-argument
         :return: iterable[DocumentWord]
         :rtype: iterable[DocumentWord]
         """
-        return _get_children(self, search_elements=["word"], cross_page=False)
+        result = []
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py:2731]: do we need to add that error if someone does a to_dict/from_dict and the parent is not available?

Comment (catalinaperalta) [reply sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py:2731]: Yes, thanks for the reminder, will update this!

Comment (catalinaperalta) [reply sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py:2735]: Done, let me know if the error message is clear enough!

**Review by kristapratico — APPROVED** [review]

**Pull request merged**